samples/vsphere/backuprestore/backup_job_crud.py

```diff
@@ -0,0 +1,203 @@
+
+        parser.add_argument('-lb', '--list-backup',
+                    required=False,
+                    action='store_true',
+                    help='Switch to list backup jobs')
+
+        parser.add_argument('-gb', '--get-backup',
+                    required=False,
+                    action='store',
+                    help='Backup Job ID to Get Information About')
+
+        parser.add_argument('-nb', '--create-backup',
+                    required=False,
+                    action='store_true',
+                    help='Switch to create backup job')                    
+
+        parser.add_argument('-cb', '--cancel-backup',
+                    required=False,
+                    action='store',
+                    help='Backup Job ID to Cancel Job')
+
+        args = parser.parse_args()
+
+        self.server = args.server
+        self.server_user = args.server_user
+        self.server_password = args.server_password
+        self.location = args.location
+        self.location_user = args.location_user
+        self.location_password = args.location_password
+        self.backup_password = args.backup_password
+        self.backup_comment = args.backup_comment
+        self.list_backup = args.list_backup
+        self.get_backup = args.get_backup
+        self.create_backup = args.create_backup
+        self.cancel_backup = args.cancel_backup
+
+        # Connect to vAPI services
+        self.stub_config = vapiconnect.connect(
+                                    host=self.server,
+                                    user=self.server_user,
+                                    pwd=self.server_password,
+                                    skip_verification="TRUE")
+
+        self.job_client = Job(self.stub_config)
+
+    def list_backup_job(self):
+        job_list = self.job_client.list()
+        self.print_output(job_list) 
+
+    def get_backup_job(self):
+        self.print_output(self.get_backup)
+
+    def create_backup_job(self):
+        backup_type = (self.location.split(':')[0]).upper()
+        piece = self.job_client.BackupRequest(
+                                    parts = ["common"], 
+                                    backup_password = self.backup_password, 
+                                    location_type = backup_type, 
+                                    location = self.location, 
+                                    location_user = self.location_user, 
+                                    location_password = self.location_password, 
+                                    comment = self.backup_comment)
+        
+        job_create = self.job_client.create(piece)
+        self.print_output(job_create.id) 
+
+
+    def cancel_backup_job(self):
+        job_cancel = self.job_client.cancel(self.cancel_backup)
+        self.print_output(self.cancel_backup)   
+
+    def print_output(self, job):
+        table = []
+
+        if type(job) is list: 
+            for bkupjob in job:
```

Comment:
We have some duplicated code here. Maybe we can extract the common code between if/else to a separate method?